[Forums] Implement a forum landing page for unauthenticated users

Luca-W-D opened the pull request:
Currently, an unauthenticated user who navigates to the forum page will see an empty forum even if content exists on the backend. This is because the security rules deny logged-out users read permissions. While secure, this leads to a confusing user experience. Therefore, as discussed in #252, it would be better to have a landing page that instructs unauthenticated users to log in. 

This PR (1) implements a landing page for unauthenticated users and (2) redirects unauthenticated redirects forum sub-pages via middleware. An unauthenticated user now receives the following view:
<img width="1449" alt="image" src="https://github.com/user-attachments/assets/e4219480-1e2b-4689-85f7-c26b5d1686a3" />

Luca-W-D committed:
[forums] Prevent unauthenticated users forum access

This extends the previous security system which merely denies content from
loading client-side. Now, this redirects users to the index page (where we
can present an unauthenticated landing page

Luca-W-D committed:
[forums] Implemented unauthenticated user landing page

Fixes: #252

Luca-W-D committed:
Merge branch 'dev' into forums/unauthenticated-landing-page

yizou275 commented on src/templates/forum/index.html:
```diff
@@ -5,5 +5,9 @@
-  {{ block.super }}
+  {% if user.is_authenticated %}
+    {{ block.super }}
+  {% else % }
```
Here there is an extra space between "%" and "}"? The page mistakenly shows "  {% else % } / Not logged in" explicitly when I access the page.
![image](https://github.com/user-attachments/assets/37b3423b-b7c8-4afa-8bd9-62c41c2513f8)

yizou275 reviewed (changes requested):
Hi Luca, thanks for this great work! It seems that the file changes in games folder (removal of favorite_list) are also included in this pr, but not directly related to the issue? It might be a good idea to recover these in this pr and open a separate pr for these unrelated file changes?

Luca-W-D committed:
[knowledge-base] Manual git merge & fix typo in template

Git for some reason isn't recognizing the changes that the
game team made as "more recent" than this previous PR. So,
I had to manually pull in their commits. Also, it appears
that an extra space causes a rendering issue for some devices.

Fixes: #646

Luca-W-D committed:
[knowledge-base] Pushing typo from last commit

Luca-W-D commented:
That information is super helpful– thank you! I agree that something weird is going on with the Git history. I think that – because this PR touches the /template folder, it won't pull in the more recent dev changes that added these files. I went ahead and manually pulled them in.

I also fixed that typo, I'm hoping that fixes your rendering issue!

Luca-W-D committed:
[knowledge-base] Modify UI to prevent duplicated content

Fixes: #646

Luca-W-D committed:
[forums] Repair config from git merge

At some point between the shuffling of branches, the
middleware configuration dropped.

Fixes #646

yizou275 reviewed (approved):
Yes this looks good to me now! Thank you so much for the great work!

Luca-W-D commented:
Thank you for the in-depth feedback throughout this PR, Yi! Especially catching that last misconfiguration– I would not have seen that. :)

Luca-W-D committed:
Merge branch 'dev' into forums/unauthenticated-landing-page

Luca-W-D commented:
Heart dropped for a second– if you look at the workflow history, the automated tests after the merge failed on the first iteration. I looked at the error logs, and it turns out that one of the automated tests generated two games as fake data with the same name, causing an integrity issue. Fortunately just bad luck rather than a code issue!

jacqsmei reviewed (approved):
nice job yall! and great catch Yi! lgtm

jacqsmei commented:
Issue Score: Excellent

Comments:
This was a great extra task to pick up! Nice teamwork figuring out the review. Great that it finally got done

<!-- {"score": "excellent", "minor": false} -->
